### PR TITLE
feat: add notification settings

### PR DESCRIPTION
**Pull request by mani-rsg**
- [x] Rename Layout Settings above to UI Settings
- [x]  Renamed title from Participant in View to UI settings and make participant text down to the title above the slider (made changes in [@100mslive/hms-video-react](https://github.com/100mslive/hms-video-react/issues/397))
- [x]  Add new section for Notifications with checkboxes for Peer Join/Peer Leave/New Message/Errors (made changes in [@100mslive/hms-video-react](https://github.com/100mslive/hms-video-react/issues/397))
- [x]  Allowed only the selected notifications to be sent from here - https://github.com/100mslive/100ms-web/blob/main/src/views/components/notifications/Notifications.jsx

Attached below screenshots for reference:

Screenshot 2021-10-06 at 8 30 13 PM
Screenshot 2021-10-06 at 8 30 58 PM

### CODE REVIEW

File: src/store/AppContext.js [hunk 1]
```diff
@@ -123,13 +123,22 @@ const AppContextProvider = ({
   const deepSetAppPolicyConfig = config =>
     setState(prevState => ({ ...prevState, localAppPolicyConfig: config }));
 
+  const deepSetSubscribedNotifications = notification => {
+    setState(prevState => {
+      if (notification.isSubscribed) prevState.subscribedNotifications[notification.type] = true;
+      else prevState.subscribedNotifications[notification.type] = false;
+      return {...prevState};
```
Comment (raviteja83) [inline src/store/AppContext.js]: `{
	...prevState,
 	subscribedNotifications: {
 		...prevState.subscribedNotifications, 
 		[notification.type]: notification.isSubscribed
 	}
 }`

File: src/views/components/MoreSettings.jsx [hunk 2]
```diff
@@ -84,8 +86,23 @@ export const MoreSettings = () => {
 
   const onChange = count => {
     setMaxTileCount(count);
+    console.log(count);
```
Comment (raviteja83) [inline src/views/components/MoreSettings.jsx]: remove console

File: src/views/components/notifications/Notifications.jsx [hunk 3]
```diff
@@ -14,24 +14,27 @@ import {
...
     }
     switch (notification.type) {
       case HMSNotificationTypes.PEER_JOINED:
         console.debug("[Peer Joined]", notification.data);
+        if(!subscribedNotifications.PEER_JOINED) return;
         break;
       case HMSNotificationTypes.PEER_LEFT:
         console.debug("[Peer Left]", notification.data);
+        if(!subscribedNotifications.PEER_LEFT) return;
         if (window.HMS.notifications?.peerLeft) {
```
Comment (raviteja83) [inline src/views/components/notifications/Notifications.jsx]: can you remove this part now. it is not needed as UI changes are done now. Also remove notifications from window.HMS in `FeatureFlags` component

File: src/store/AppContext.js [hunk 4]
```diff
@@ -70,6 +69,7 @@ const AppContextProvider = ({
     loginInfo: initialLoginInfo,
     maxTileCount: 9,
     localAppPolicyConfig: {},
+    subscribedNotifications: { "PEER_JOINED": true, "PEER_LEFT": true, "NEW_MESSAGE": true, "ERROR": true }
```
Comment (raviteja83) [inline src/store/AppContext.js]: make peer_join and peer_left  as false by default

**Review by raviteja83 — CHANGES_REQUESTED** [review]

**Comment by triptu** [comment]
@mani-rsg can you please send your resume as well to tushar@100ms.live :) ?

**Comment by raviteja83** [comment]
@mani-rsg can you update @100mslive/hms-video-react version

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**